### ISSUE TITLE
Add Enum instead of String type for Events and States

### ISSUE THREAD

**Issue by makarof87** [statefulj/statefulj#9]
Is it a big deal to change types for events and states to enums? So it will be more consistent to work with framework.

**Comment by developer-x** [comment]
I agree that Enums are more appropriate for defining a finite set of values or states.  However, I chose to deal with Strings because the "state" values are used a keys to hashsets and are updated directly in the database via various ORM implementations (hibernate and mongo).  Since String are final classes, hashCode and equals cannot be overriden.  Additionally, persisting Strings to the database vs. Enums is a bit more straightforward.  So I decided to go with Strings because I felt it would cause less problems when integrating StatefulJ with the various db/orm technologies.

Now, I haven't spent much time investigating supporting Enums - if there was a pull request demonstrating that working with Enums vs. Strings wouldn't be problematic, then I'd be happy to consider it.
